source/Ship.cpp

```diff
@@ -1913,6 +1913,16 @@ double Ship::JumpFuel() const
 
 
 
+double Ship::JumpFuelMissing() const
+{
+	// Used for smart refuelling: transfer only as much as really needed
+	// includes checking if fuel cap is high enough at all
+	if(!JumpsRemaining() && attributes.Get("fuel capacity") > JumpFuel())
+		return JumpFuel() - fuel;
```

Comment:
The 2nd check shall prevent to return any Missing fuel if the ship cant hold enough fuel for the specific drive.
I.e. if you have a ship with e.g. fuel capaity 150 and install a "jump drive" this function shall always return 0.

Good point with the control path.